src/cr/cube/cube_slice.py

```diff
@@ -150,3 +150,33 @@ def ca_main_axis(self):
             return 1 - ca_ind
         else:
             return None
+
+    def labels(self, hs_dims=None, prune=False):
+        '''Get labels for the cube slice, and perform pruning by slice.'''
+        labels = self._cube.labels(include_transforms_for_dims=hs_dims)[-2:]
+        if not prune:
+            return labels
+
+        def prune_dimension_labels(labels, prune_indices):
+            '''Get pruned labels for single dimension, besed on prune inds.'''
+            labels = [
+                label for label, prune in zip(labels, prune_indices)
+                if not prune
+            ]
+            return labels
+
+        prune_indices = self.prune_indices(transforms=hs_dims)
+        labels = [
+            prune_dimension_labels(dim_labels, dim_prune_inds)
```

Comment:
Why not just inline this? The function logic is trivial, a single expression.

The over-optimizer in me bristles at the function call overhead and the expense of creating an inline function that's not a closure. :)

But the mental overhead of having to consider the function and this expression separately is the bigger cost, IMO.

Reply:
Done